packages/react-compose/src/resolveClasses.test.ts

```diff
@@ -0,0 +1,27 @@
+import { resolveClasses } from './resolveClasses';
+
+describe('resolveClasses', () => {
+  it('resolves className', () => {
+    expect(
+      resolveClasses([], { state: { className: 'cn' }, slotProps: { root: { className: 'root' } }, slots: {} }),
+    ).toEqual({
+      state: { className: 'cn' },
+      slotProps: { root: { className: 'root cn' } },
+      slots: {},
+    });
+  });
+
+  it('can distribute classes to slotProps', () => {
+    expect(
+      resolveClasses([{ root: 'root' }, { slot1: 'slot1' }, { unrecognized: 'ignored' }], {
```

Comment:
i think we can use more tests here:
1. when `classes` contain overlapped prop: 
  `resolveClasses([{ root: 'root' }, { slot1: 'slot1' }, { slot1: 'slot2' }]`
2. when a class is a function
3. duplicated classes:
  `resolveClasses([{ root: 'root' }, { slot1: 'slot1' }, { slot1: 'slot1' }]`

Reply:
done